src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,27 @@
+package seedu.addressbook.data.person;
+
+public abstract class Contact {
+
+    public String value;
+    protected boolean isPrivate;
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public int hashCode() {
+        return value.hashCode();
+    }
```

Comment:
In general, we override `equals` together with `hashCode`, you can read up more on why this is so :)